pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -5690,49 +5794,49 @@ def BeneficialReuseWaterQuality(b, o, w, t):
     # The first input is output stream (1) from the intermediate step.
     # The second is outgoing flow from the storage tank.
 
-    def CompletionsPadIntermediateWaterQuality(b, p, w, t):
+    def CompletionsPadIntermediateWaterQuality(b, p, qc, t):
         constraint = sum(
-            b.parent_block().v_F_Piped[n, p, t] * b.v_Q[n, w, t]
+            b.parent_block().v_F_Piped[n, p, t] * b.v_Q[n, qc, t]
             for n in b.parent_block().s_N
             if b.parent_block().p_NCA[n, p]
         ) + sum(
-            b.parent_block().v_F_Piped[p_tilde, p, t] * b.v_Q[p_tilde, w, t]
+            b.parent_block().v_F_Piped[p_tilde, p, t] * b.v_Q[p_tilde, qc, t]
             for p_tilde in b.parent_block().s_PP
             if b.parent_block().p_PCA[p_tilde, p]
         ) + sum(
-            b.parent_block().v_F_Piped[s, p, t] * b.v_Q[s, w, t]
+            b.parent_block().v_F_Piped[s, p, t] * b.v_Q[s, qc, t]
             for s in b.parent_block().s_S
             if b.parent_block().p_SCA[s, p]
         ) + sum(
-            b.parent_block().v_F_Piped[p_tilde, p, t] * b.v_Q[p_tilde, w, t]
+            b.parent_block().v_F_Piped[p_tilde, p, t] * b.v_Q[p_tilde, qc, t]
             for p_tilde in b.parent_block().s_CP
             if b.parent_block().p_CCA[p_tilde, p]
         ) + sum(
-            b.parent_block().v_F_Piped[r, p, t] * b.v_Q[r, w, t]
+            b.parent_block().v_F_Piped[r, p, t] * b.v_Q[r + treated_water_label, qc, t]
```

Comment:
This is essentially the same question as I put in the documentation, but why use `treated_water_label` here?

Reply:
@tarnold17 To expand on @melody-shellman response, it's worth noting that the treated water (with quality of  Q[r + treated_water_label, qc, t]) - as opposed to the treatment feed (with quality of Q[r, qc, t])) - is sent to completion pads for reuse. This information is also reflected in the Treated Water Balance constraint:

    def TreatedWaterRule(model, r, t):
        constraint = (
            model.v_F_TreatedWater[r, t]
            == sum(model.v_F_Piped[r, p, t] for p in model.s_CP if model.p_RCA[r, p])
            + sum(model.v_F_Piped[r, s, t] for s in model.s_S if model.p_RSA[r, s])
            + model.v_F_DesalinatedWater[r, t]
        )
        return process_constraint(constraint)